src/WopiValidator/Program.cs

```diff
@@ -65,6 +65,41 @@ private static ExitCode Execute(Options options)
 			// get run configuration from XML
 			IEnumerable<TestExecutionData> testData = ConfigParser.ParseExecutionData(options.RunConfigurationFilePath, options.TestCategory);
 
+			// Set header "X-WOPI-ApplicationId" for ReadSecureStore request.
+			// Set header "X-WOPI-UsingRestrictedScenario" for GetRestrictedLink and RevokeRestrictedLink request.
+			if (!string.IsNullOrEmpty(options.UsingRestrictedScenario)
```

Comment:
This logic shouldn't be here. It should be in the RequestFactory for the particular type of request.